### DISCUSSION TITLE
IKE connection fail: INFORMATIONAL_V1 request [ N(PLD_MAL) ]

### DISCUSSION THREAD

**Discussion by misch42** [strongswan/strongswan#1701] [General]
Hi,

we want to set up a IKEv1 (yes, I know!) site-2-site connection to a Fortinet. The confi is more or less copied from the IKKv1 examples. But the connection fails with the following log:

[IKE] initiating Main Mode IKE_SA remote[57] to r.r.r.r
[ENC] generating ID_PROT request 0 [ SA V V V V V ]
[NET] sending packet: from l.l.l.l[500] to r.r.r.r[500] (180 bytes)
[NET] received packet: from r.r.r.r[500] to l.l.l.l[500] (124 bytes)
[ENC] parsed ID_PROT response 0 [ SA V V ]
[IKE] received FRAGMENTATION vendor ID
[IKE] received NAT-T (RFC 3947) vendor ID
[CFG] selected proposal: IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_4096
[ENC] generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
[NET] sending packet: from l.l.l.l[500] to r.r.r.r[500] (652 bytes)
[NET] received packet: from r.r.r.r[500] to l.l.l.l[500] (660 bytes)
[ENC] parsed ID_PROT response 0 [ KE No NAT-D NAT-D V ]
[IKE] received DPD vendor ID
[ENC] generating ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
[NET] sending packet: from l.l.l.l[500] to r.r.r.r[500] (108 bytes)
[NET] received packet: from r.r.r.r[500] to l.l.l.l[500] (40 bytes)
[ENC] parsed INFORMATIONAL_V1 request 875135164 [ N(PLD_MAL) ]
[ENC] ignoring unprotected INFORMATIONAL from r.r.r.r
[IKE] message verification failed
[IKE] ignore malformed INFORMATIONAL request
[IKE] INFORMATIONAL_V1 request with message ID 875135164 processing failed

Any ideas why the connection fails? Or should we look for the reason in the logs of the other side?

Michael.

**Comment by tobiasbrunner** [comment]
> we want to set up a IKEv1 (yes, I know!)

Now [it's even official](https://datatracker.ietf.org/doc/html/rfc9395) ;)

> Any ideas why the connection fails?

Incorrect PSK would be my guess.

**Comment by misch42** [comment]
The strongswan logs also say if the other side initiates the connection: 

charon[2632910]: 04[NET] received packet: from r.r.r.r[500] to l.l.l.l[500] (172 bytes)
charon[2632910]: 04[ENC] parsed ID_PROT request 0 [ SA V V V ]
charon[2632910]: 04[IKE] received NAT-T (RFC 3947) vendor ID
charon[2632910]: 04[IKE] received FRAGMENTATION vendor ID
charon[2632910]: 04[IKE] received Checkpoint Firewall vendor ID
charon[2632910]: 04[IKE] r.r.r.r is initiating a Main Mode IKE_SA
charon[2632910]: 04[IKE] r.r.r.r is initiating a Main Mode IKE_SA
charon[2632910]: 04[CFG] selected proposal: IKE:AES_CBC_128/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_4096
charon[2632910]: 04[ENC] generating ID_PROT response 0 [ SA V V V V ]
charon[2632910]: 04[NET] sending packet: from l.l.l.l[500] to r.r.r.r[500] (164 bytes)
charon[2632910]: 06[NET] received packet: from r.r.r.r[500] to l.l.l.l[500] (660 bytes)
charon[2632910]: 06[ENC] parsed ID_PROT request 0 [ KE No NAT-D NAT-D V ]
charon[2632910]: 06[IKE] received DPD vendor ID
charon[2632910]: 06[ENC] generating ID_PROT response 0 [ KE No NAT-D NAT-D ]
charon[2632910]: 06[NET] sending packet: from l.l.l.l[500] to r.r.r.r[500] (652 bytes)
charon[2632910]: 05[NET] received packet: from r.r.r.r[500] to l.l.l.l[500] (92 bytes)
charon[2632910]: 05[ENC] invalid ID_V1 payload length, decryption failed?
charon[2632910]: 05[ENC] could not decrypt payloads
charon[2632910]: 05[IKE] message parsing failed
charon[2632910]: 05[ENC] generating INFORMATIONAL_V1 request 3579229484 [ HASH N(PLD_MAL) ]
charon[2632910]: 05[NET] sending packet: from l.l.l.l[500] to r.r.r.r[500] (92 bytes)
charon[2632910]: 05[IKE] ID_PROT request with message ID 0 processing failed

**Comment by misch42** [comment]
Same connection worked two days ago. I did not change the PSK.

**Reply by tobiasbrunner** [reply]
Then the other end did ;)

**Reply by khanhkenit** [reply]
how it worked bro. I have the same situation

**Comment by misch42** [comment]
On 16.05.23 16:23, Tobias Brunner wrote:
>
> Then the other end did ;)
>

Seems they did.
